// analytics-plugin-simple-analytics/lib/analytics-plugin-simple-analytics.browser.es.js
/* global sa */

/**
 * Simple Analytics plugin
 * @link https://getanalytics.io/plugins/simple-analytics/
 * @link https://simpleanalytics.com/
 * @return {object} Analytics plugin
 * @example
 *
 * simpleAnalyticsPlugin()
 */
function simpleAnalyticsPlugin$1() {
  var pluginConfig = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  var isLoaded = false; // Allow for userland overides of base methods

  return {
    NAMESPACE: 'simple-analytics',
    config: pluginConfig,
    // https://docs.simpleanalytics.com/script
    initialize: function initialize(_ref) {
      var config = _ref.config;
      var src = 'https://cdn.simpleanalytics.io/hello.js';
      var script = document.createElement('script');
      script.type = 'text/javascript';
      script.async = true;
      script.src = "".concat(src); // Append the script to the DOM

      var el = document.getElementsByTagName('script')[0];
      el.parentNode.insertBefore(script, el); // Resolve the promise once the script is loaded

      script.addEventListener('load', function () {
        isLoaded = true;
      }); // Catch any errors while loading the script

      script.addEventListener('error', function () {
        throw new Error("".concat(src, " failed to load."));
      }); // Todo load analytics script via users subdomain for custom event tracking
    },

    /* https://docs.simpleanalytics.com/events
    track: ({ payload }) => {
      if (typeof sa === 'undefined') return false
      sa(payload.event)
    },
    */

    /* Verify script loaded */
    loaded: function loaded() {
      return !!isLoaded;
    }
  };
}

var index = simpleAnalyticsPlugin$1;

export default index;
